File: 0019.remove_nth_from_end/solution.py
from utils import ListNode


class Solution:
    def remove_nth_from_end(self, head: ListNode, n: int) -> ListNode:
        if not head:
            return
        ans = ListNode(0)
        ans.next = head
        fast, slow = ans, ans
        for _ in range(n + 1):
            fast = fast.next
        while fast.next:
            fast = fast.next
            slow = slow.next
        slow.next = slow.next.next
        return ans.next

File: 0019.remove_nth_from_end/solution.go
package leetcode0019

import "github.com/Ackerr/Algorithms/utils"

func removeNthFromEnd(head *utils.ListNode, n int) *utils.ListNode {
	if head == nil {
		return nil
	}
	fast := head
	for i := 0; i < n; i++ {
		fast = fast.Next
	}
	if fast == nil {
		return head.Next
	}
	slow := head
	for fast.Next != nil {
		fast = fast.Next
		slow = slow.Next
	}
	slow.Next = slow.Next.Next
	return head
}
